iam.tf

```diff
@@ -33,3 +33,86 @@ resource "aws_iam_instance_profile" "this" {
   name = local.namespace
   role = var.create_iam_role ? aws_iam_role.this[0].name : var.iam_role_arn
 }
+
+data "aws_iam_policy_document" "lambda_policy" {
+  # Allow the Lambda to write CloudWatch Logs.
+  statement {
+    effect = "Allow"
+    actions = [
+      "logs:CreateLogStream",
+      "logs:PutLogEvents",
+    ]
+
+    resources = ["${aws_cloudwatch_log_group.log_group.arn}:*"]
+  }
+
+  # Allow the Lambda to put X-Ray traces.
+  statement {
+    effect = "Allow"
+    actions = [
+      "xray:PutTraceSegments",
+      "xray:PutTelemetryRecords",
+    ]
+
+    resources = ["*"]
+  }
+
+  # Allow the Lambda to DescribeAutoScalingGroups, DetachInstances and SetDesiredCapacity
+  # on the AutoScalingGroup.
+  statement {
+    effect = "Allow"
+    actions = [
+      "autoscaling:DetachInstances",
+      "autoscaling:SetDesiredCapacity",
+    ]
+
+    resources = [module.asg.autoscaling_group_arn]
+  }
+
+  statement {
+    effect    = "Allow"
+    actions   = ["autoscaling:DescribeAutoScalingGroups"]
+    resources = ["*"]
```

Comment:
Can we limit it to `module.asg.autoscaling_group_arn` as above?

Reply:
Done